sotodlib/site_pipeline/update_obsdb.py

```diff
@@ -0,0 +1,160 @@
+    tback = tnow - recency*86400
+    #Find folders that are book-like and recent
+    for dirpath,_, _ in os.walk(base_dir):
+        last_mod = max(os.path.getmtime(root) for root,_,_ in os.walk(dirpath))
+        if last_mod<tback:#Ignore older directories
+            continue
+        if os.path.exists(os.path.join(dirpath, "M_index.yaml")):
+            #Looks like a context file
+            bookcart.append(dirpath)
+    #Check the books for the observations we want
+
+    for bookpath in bookcart:
+        if check_meta_type(bookpath) in accept_type:
+            """ obsfiledb creation
+            The extra files give us a bit of trouble. We'll assume that index,
+            book, and bookbinder_log are always here, and edit our config file
+            to cover the rest as extra files.
+            """
+            book_file_list = os.listdir(bookpath)
+            book_file_list.remove("M_index.yaml")
+            book_file_list.remove("M_book.yaml")
+            book_file_list.remove("Z_bookbinder_log.txt")
+            book_file_list = [file_name for file_name in book_file_list if not file_name.endswith(".g3")]
+            if len(book_file_list)>0:
+                configs["extra_extra_files"] += book_file_list
+                with open(config, "w") as config_file:
+                    yaml.dump(configs, config_file)
+                checkbook(bookpath, config, add=True)
+                configs["extra_extra_files"] = ["Z_bookbinder_log.txt"]
+                with open(config, "w") as config_file:
+                    yaml.dump(configs, config_file)
+
+            else:
+                checkbook(bookpath, config, add=True)
+
+            index = yaml.safe_load(open(os.path.join(bookpath, "M_index.yaml"), "rb"))
+            obs_id = index.pop("book_id")
+            tags = index.pop("tags")
+            detsets = index.pop("detsets")
+
+            if "cols" in configs:
+                very_clean = {col:index[col] for col in iter(configs["cols"]) if col in index}
+            else:
+                col_list = []
+                clean = {key:val for key, val in index.items() if val is not None}
+                very_clean = {key:val for key, val in clean.items() if type(val) is not list}
+                for key, val in very_clean.items():
+                    col_list.append(key+" "+type(val).__name__)
+                bookcartobsdb.add_obs_columns(col_list)
+
+            if tags != ([] or [""]):
+                bookcartobsdb.update_obs(obs_id, very_clean, tags=tags)
+            else:
+                bookcartobsdb.update_obs(obs_id, very_clean)
+           
+        else:
+            bookcart.remove(bookpath)
+    if "obsdb" in configs:
+        bookcartobsdb.to_file(configs["obsdb"])
+    else:
+        bookcartobsdb.to_file("obsdb_from{}_to{}".format(tback, tnow))
+
+
+def get_parser(parser=None):
+    if parser is None:
+        parser = argparse.ArgumentParser()
+    parser.add_argument('--base_dir', type=str, 
```

Comment:
At a minimum we need base-dir to be a required argument, because we can't add things without it. But I think I would actually prefer for base-dir to be something specified in the config file because it shouldn't change once we've set up databases.

Reply:
You're right, I'll put it in the config file